libs/ml/tests/ml/model/sequential_test.cpp

```diff
@@ -122,32 +123,95 @@ TYPED_TEST(ModelsTest, adagrad_sequential)
 
   // TODO (1556) - ADAGRAD not currently working
   //  using DataType  = typename TypeParam::Type;
-  //  ASSERT_TRUE(RunTest<TypeParam>(fetch::ml::optimisers::OptimiserType::ADAGRAD,
+  //  ASSERT_TRUE(RunTest<TypeParam>(fetch::ml::OptimiserType::ADAGRAD,
   //  static_cast<DataType>(1e-1)));
 }
+
 TYPED_TEST(ModelsTest, adam_sequential)
 {
   using DataType = typename TypeParam::Type;
-  ASSERT_TRUE(RunTest<TypeParam>(fetch::ml::optimisers::OptimiserType::ADAM,
-                                 static_cast<DataType>(1e-5),
-                                 static_cast<typename TypeParam::Type>(0.1)));
+  ASSERT_TRUE(RunTest<TypeParam>(fetch::ml::OptimiserType::ADAM, static_cast<DataType>(1e-3),
+                                 static_cast<DataType>(1e-2), 10));
```

Comment:
Does this test fail with the old values for the tolerance? Could the tolerance be expressed here in terms of the datatype tolerance?

Reply:
It's not the same tolerance, because it is ground truth/prediction difference.

Reply:
It doesn't depend on datatype

Reply:
Even so, this tolerance is quite high!
Is this because default learning rate is bad or because number of training runs is too small? either way It would be preferable to have these hyperparameters set appropriately via input parameters to RunTest function call rather than compensate for bad hyperparameters with high tolerance.